client/src/Features/MicrosoftLearn/MicrosoftLearnFilterComponentUtils.tsx

```diff
@@ -0,0 +1,151 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useStore } from '../../Stores/Core';
+import { useObserver } from 'mobx-react-lite';
+import React from 'react';
+import {
+  getProductsToDisplay,
+  getTypesToDisplay,
+  getLevelsToDisplay,
+  getRolesToDisplay
+} from './MicrosoftLearnFilterUtils';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { MicrosoftLearnFilterComponent } from './MicrosoftLearnFilterComponent';
+import { FontWeights } from '@fluentui/react';
+import { IThemeOnlyProps, SimpleComponentStyles } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { themedClassNames } from '../../Core/Utils/FluentUI';
+
+export type FilterComponentStyles = SimpleComponentStyles<
+  'root' | 'title' | 'search' | 'optionsList' | 'subOptionsList' | 'filterItem'
+>;
+
+type FilterComponentTypes = FilterType.products | FilterType.roles | FilterType.levels | FilterType.types;
+
+export const FilterComponent = (props: { type: FilterComponentTypes; name: string }) => {
+  const { filterStore, catalog } = useStore('microsoftLearnStore');
+  return useObserver(() => {
+    
+    const getDisplayOptions = () => {
```

Comment:
Thanks for this! Can we also get this function out of the return statement?

Reply:
Yeah, I did that, I guess it was after I committed. Will update the remote shortly.